.github/workflows/ci-pipeline.yml

```diff
@@ -14,3 +14,33 @@ jobs:
         run: docker run todo-app-test tests/integration_tests
       - name: End to end tests 
         run:  docker run -e TRELLO_API_KEY=${{ secrets.TRELLO_API_KEY }} -e TRELLO_API_TOKEN=${{ secrets.TRELLO_API_TOKEN }} -e TRELLO_ORGANIZATION_ID=${{ secrets.TRELLO_ORGANIZATION_ID }} todo-app-test tests_e2e
+  deploy:
+    name: Publish and deploy
+    needs: build
+    if: (github.event_name == 'push' && github.ref == 'refs/heads/main')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: kirstyland
+          password: ${{ secrets.DOCKER_HUB_KEY }}
+      - name: Build docker image
+        run: docker build --target production --tag kirstyland/todo-app:latest .
+      - name: Tag docker image
+        run: docker tag kirstyland/todo-app:latest kirstyland/todo-app:$GITHUB_SHA

```

Comment:
You should be able to set multiple tags within your build command instead of needing a separate `docker tag` command

You could even do all three tags, but fair enough if you want to keep the Heroku-specific stuff separate.